bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,575 @@
+<plist version="1.0">
+  <dict>
+    <key>kubernetes</key>
+    <dict>
+      <key>version</key>
+      <string>moose-head</string>
+    </dict>
+  </dict>
+</plist>
+EOF
+}
+
+@test 'generates plist output from a command-line argument' {
+    run rdctl create-profile --output plist --body "$(simple_json_data)"
+    assert_moose_head_plist_output
+}
+
+@test 'generates plist output from a file' {
+    run rdctl create-profile --output plist --input <(simple_json_data)
+    assert_moose_head_plist_output
+}
+
+@test 'verify plutil is ok with the generated plist output from input file' {
+    if ! is_macos; then
+        skip "Test requires the plist utility and only works on macOS"
+    fi
+    run rdctl create-profile --output plist --input <(simple_json_data)
+    assert_success
+    plutil -s - <<<"$output"
+}
+
+assert_complex_plist_output() {
+    assert_success
+    assert_output - <<'EOF'
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>containerEngine</key>
+    <dict>
+      <key>allowedImages</key>
+      <dict>
+        <key>patterns</key>
+        <array>
+          <string>abc</string>
+          <string>ghi</string>
+          <string>def</string>
+        </array>
+      </dict>
+    </dict>
+    <key>kubernetes</key>
+    <dict>
+      <key>enabled</key>
+      <false/>
+    </dict>
+    <key>WSL</key>
+    <dict>
+      <key>integrations</key>
+      <dict>
+        <key>first</key>
+        <true/>
+        <key>second</key>
+        <false/>
+      </dict>
+    </dict>
+  </dict>
+</plist>
+
+EOF
+}
+
+@test 'plist-encodes multi-string values and maps from a file' {
+    run rdctl create-profile --output plist --input <(complex_json_data)
+    assert_complex_plist_output
+}
+
+@test 'plist-encodes multi-string values and maps from a json string' {
+    run rdctl create-profile --output plist --body "$(complex_json_data)"
+    assert_complex_plist_output
+}
+
+json_with_special_chars() {
+    echo '{ "application": {
+            "extensions": {
+                "allowed": {
+                  "enabled": false,
+                  "list": ["less-than:<", "greater:>", "and:&", "d-quote:\"", "emoji:😀"]
+                },
+                "installed": {
+                    "key-with-less-than: <": true,
+                    "key-with-ampersand: &": true,
+                    "key-with-greater-than: >": true,
+                    "key-with-emoji: 🐤": false
+                }
+            }
+        },
+        "containerEngine": {
+          "name": "small-less-<-than"
+        }
+}'
+}
+
+# Actual output-testing of this input is done in `plist_test.go` -- the purpose of this test is to just
+# make sure that we're generating compliant data.
+
+@test 'verify converted special-char input is escaped and satisfies plutil' {
```

Comment:
Why is there no corresponding test for `reg` output? We can't verify it with `reg.exe`, but we could compare the output to a here-doc, to make sure the data is represented correctly.

Reply:
Full-output comparison already done in `assert_full_setting_registry_output`.

And we can verify that `reg` is happy with the generated output, but we have to use a destination other than `HKCU/SOFTWARE/Profiles`, because you need to be an admin to update that. But why not...

Reply:
> And we can verify that `reg` is happy with the generated output,

That's not what I meant. I meant why is there a test that "special characters" make it correctly into plist output, but no corresponding test to make it into reg output. Or maybe I missed it?

Reply:
Ah right. Now that I have reg-file checking let's see if it's actually acceptable.

…

Surprise ... it is.